src/Kiota.Builder/Refiners/CSharpRefiner.cs

```diff
@@ -42,36 +40,61 @@ public override void Refine(CodeNamespace generatedCode)
         }
         private static void MakeEnumPropertiesNullable(CodeElement currentElement) {
             if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model))
-                currentClass.GetChildElements(true)
-                            .OfType<CodeProperty>()
+                currentClass.Properties
                             .Where(x => x.Type is CodeType propType && propType.TypeDefinition is CodeEnum)
                             .ToList()
                             .ForEach(x => x.Type.IsNullable = true);
             CrawlTree(currentElement, MakeEnumPropertiesNullable);
         }
         private static void AddParsableInheritanceForModelClasses(CodeElement currentElement) {
-            if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model)) {
-                var declaration = currentClass.StartBlock as CodeClass.Declaration;
-                declaration.Implements.Add(new CodeType(currentClass) {
+            if(currentElement is CodeClass currentClass &&
+                currentClass.IsOfKind(CodeClassKind.Model) &&
+                currentClass.StartBlock is CodeClass.Declaration declaration) {
+                declaration.AddImplements(new CodeType {
                     IsExternal = true,
                     Name = $"IParsable",
                 });
-                declaration.Usings.Add(new CodeUsing(currentClass) {
-                    Name = "Microsoft.Kiota.Abstractions.Serialization"
+                (currentClass.Parent is CodeClass parentClass &&
+                parentClass.StartBlock is CodeClass.Declaration parentDeclaration ? 
+                    parentDeclaration :
+                    declaration)
+                    .AddUsings(new CodeUsing {
```

Comment:
I believe we might only need to do this for `CodeClassKind.Model` as well. Yes?

Reply:
not necessary, technically request parameters could be parsable and we might need to serialize them.

Reply:
Do we have instances of that so far? I haven't spotted any so far...

Reply:
not in microsoft graph as far as I can remember, but again, Kiota is not just for Microsoft Graph :)

Reply:
Ahh. Right you are.